lib/elixir/test/elixir/stream_test.exs

```diff
@@ -1086,6 +1107,9 @@ defmodule StreamTest do
 
     assert Stream.chunk_every([0, 1, 2, 3], 2) |> Stream.zip() |> Enum.to_list() ==
              [{0, 2}, {1, 3}]
+
+    stream = %HaltAcc{acc: 1..3}
```

Comment:
Do we need to use a separate module? Couldn't we use a range + Stream.take for this test?

Reply:
If I understood correctly when this happens range + Stream.take would return only the last element halted, so it matches with the previous clause `{_, [elem]}`. In my case, the halt tuple won't match because it has 3 elements

Reply:
Oh, I see. Perfect then!